### ISSUE TITLE
Make Context Input requirement checks syntactic

### ISSUE THREAD

**Issue by PhilippGackstatter** [iotaledger/iota.go#693]
Make Context Input requirement checks syntactic.

For any feature or unlock condition that requires a Commitment, Block Issuance Credit or Reward Input we want to check syntactically whether that requirement is fulfilled. Currently, this is done at the semantic validation stage.

We can implement that as: `func OutputsSyntacticalContextInputs() ElementValidationFunc[Output]`.

Note that we do not have to implement the check that a BIC or Reward Input requires a Commitment Input, since that is already checked in `ContextInputsCommitmentInputRequirement`.

Finally, also update the TIPs accordingly to move the check from the semantic to the syntactic section.

**Comment by PhilippGackstatter** [comment]
- Whether Reward Inputs are needed is context-dependent. In other words, it can only be done at the semantic stage.
    - Delegation Output require a Reward Input when they're destroyed. Checking whether a chain is destroyed is inherently semantic. (It's impossible to look at the output side alone and see that anything is destroyed, because by definition it's not there.)
    - Staking Feature removal requires a Reward Input, but that too can only be checked at the semantic stage.
    - So Reward Input checks remain at the semantic stage.
- Whether a Block Issuance Credit Input is required depends on the input: An implicit account or a block issuer account require a BIC Input, but since we don't have the resolved inputs at the syntactic stage, we cannot make the BIC Input check there either.
- Whether a Commitment Input is required:
    - Can be checked syntactically if there is a Staking Feature on the output side of an account. Then a Commitment Input must be present (in order to set the epoch indices which are calculated from the commitment input). We must still check _semantically_ that it is present when the Staking Feature is only on the **input side**, i.e. when a Staking Feature is removed.
        - Same thing for Block Issuer Features on the output side (we need the commitment to check that expiry slot is sufficiently far in the future).
    - For Delegation Outputs the situation is similar: If it is present on the input or output side, we need a Commitment Input. A subset of that - Delegation Output on output side - can be checked syntactically, but once again, not the input side.
    - For Implicit Accounts we don't require a Commitment Input when they're created because we implicitly set the Expiry Slot to "max slot". And when they're on the input side we cannot check that syntactically.

So, to summarize, Reward and BIC Input cannot be checked syntactically because whether they are needed or not depends on the input, not the output.
A subset of the cases where a Commitment Input is required can be checked syntactically. So the question is: Should we?

We've already seen from user examples that Context Inputs are tricky and tend to be forgotten about, so the earlier we can catch these issues the better. Hence I think it's still worth it to make that subset of checks syntactic.
